### PR TITLE
[docs] Initial docs reorg

### PR DESCRIPTION
**Pull request by econchick**
This is the initial start to cleaning up/re-organizing our docs. More to come :) 

The commit messages themselves should be pretty self-explanatory. There are no changes to actual content except for small updates to titles/headings. 

Please look at the RTDs built from the PR since changes are more visual.

## Checklist for PR author(s)

- [ ] Format the pull request title like `[cli] Fixes bugs in 'klio job fake-cmd'`.
- [ ] Changes are covered by unit tests (no major decrease in code coverage %) and/or integration tests.
- [ ] Document any relevant additions/changes in the appropriate spot in `docs/src`.

### CODE REVIEW

File: docs/src/quickstart/hello_klio_batch.rst [hunk 1]
```diff
@@ -1,113 +1,3 @@
-.. _hello-klio-batch:
+:orphan:
```
Comment (econchick) [inline docs/src/quickstart/hello_klio_batch.rst:1]: (meta comment) Having both "quick start" and "user guide" as separate things could be confusing to others. As well, the quickstart has installation instructions, which could be argued should (also) be in a user guide. So I just moved the quickstart docs under the user guide.

Comment (econchick) [reply docs/src/quickstart/hello_klio_batch.rst:1]: Also, no content was changed, just moved

File: docs/src/conf.py [hunk 2]
```diff
@@ -233,4 +233,6 @@ def find_version(*file_paths):
     "userguide/intro": "index.html",
     # Move quickstart into userguide
     "quickstart/*": "/userguide/${source}.html",
+    # Move io under pipelines
```
Comment (econchick) [inline docs/src/conf.py:236]: The IO sidebar nav was only one page; and to me, it makes more sense to be nested under Pipelines.

Comment (econchick) [reply docs/src/conf.py:236]: (again, no content was changed for the file that was moved)

**Comment by econchick** [comment]
I can't figure out what's causing `make linkcheck` to fail - it seems pretty deep into sphinx. I'm investigating :)

**Comment by econchick** [comment]
Found the build issue - it's a bug in the newly-added `sphinx-reredirect` extension added. I've submitted a [PR](https://gitlab.com/documatt/sphinx-reredirects/-/merge_requests/1) for a fix, but I'm going to try to figure out a work-around in case this isn't merged & released.

**Comment by econchick** [comment]
Figured out a workaround that can be easily reverted if/when the bug is addressed

**Review by fallonchen — APPROVED** [review]
🧹 These changes make sense to me!

**Comment by econchick** [comment]
thanks @fallonchen !

**Pull request merged**